src/main/clojure/jarl/parser.clj

```diff
@@ -9,226 +9,170 @@
 (declare make-blocks)
 (declare make-stmts)
 
-(defn make-ArrayAppendStmt [stmt-info]
+(defn make-ArrayAppendStmt [{:strs [array value]}]
   (log/debug "making ArrayAppendStmt stmt")
-  (let [array-index (get stmt-info "array")
-        value-index (get stmt-info "value")]
-    (fn [state]
-      (eval/eval-ArrayAppendStmt array-index value-index state))))
+  (fn [state]
+    (eval/eval-ArrayAppendStmt array value state)))
 
-(defn make-AssignIntStmt [stmt-info]
+(defn make-AssignIntStmt [{:strs [value target]}]
   (log/debug "making AssignIntStmt stmt")
-  (let [value (get stmt-info "value")
-        target (get stmt-info "target")]
-    (fn [state]
-      (eval/eval-AssignIntStmt value target state))))
+  (fn [state]
+    (eval/eval-AssignIntStmt value target state)))
 
-(defn make-AssignVarStmt [stmt-info]
+(defn make-AssignVarStmt [{:strs [target source]}]
   (log/debug "making AssignVarStmt stmt")
-  (let [source-index (get stmt-info "source")
-        target (get stmt-info "target")]
-    (fn [state]
-      (eval/eval-AssignVarStmt source-index target state))))
+  (fn [state]
+    (eval/eval-AssignVarStmt source target state)))
 
-(defn make-AssignVarOnceStmt [stmt-info]
+(defn make-AssignVarOnceStmt [{:strs [target source]}]
   (log/debug "making AssignVarOnceStmt stmt")
-  (let [source-index (get stmt-info "source")
-        target (get stmt-info "target")]
-    (fn [state]
-      (eval/eval-AssignVarOnceStmt source-index target state))))
+  (fn [state]
+    (eval/eval-AssignVarOnceStmt source target state)))
 
-(defn make-BlockStmt [stmt-info]
+(defn make-BlockStmt [{:strs [blocks] :as stmt-info}]
```

Comment:
Destructuring in clojure is new to me. `stmt-info` is the original argument, and not an alias for `blocks`?